server/services/user.js

```diff
@@ -90,6 +90,19 @@ function removeGroup(user, groupName) {
     .then(validatedName => user.removeGroup(validatedName));
 }
 
+function getGroupsForUser(user) {
+  return new Promise((resolve, reject) => {
+    User.findOne(user)
+    .populate('activeGroups')
+    .exec((err, u) => {
```

Comment:
exec doesn't return a promise http://stackoverflow.com/questions/31549857/mongoose-what-does-the-exec-function-do and I want to return the value in a specific format. The resolve helps me do that.